contracts/proxy/StablePoolProxy.sol

```diff
@@ -40,11 +41,19 @@ contract StablePoolProxy is Initializable, OwnableUpgradeSafe {
 
     event Initialized(address indexed thisAddress);
 
-    function initialize() public initializer {
+    function initialize(address _advisor) public initializer {
         OwnableUpgradeSafe.__Ownable_init();
+        OwnableUpgradeSafe.transferOwnership(_advisor);
```

Comment:
Why are you passing in _advisor here? Doesn't the caller who initializes these pools get ownership? Can we add a test either proving or disproving this?